proto/pulumi/provider.proto

```diff
@@ -594,35 +603,78 @@ message DiffResponse {
     }
 }
 
+// `CreateRequest` is the type of requests sent as part of a [](pulumirpc.ResourceProvider.Create) call.
 message CreateRequest {
-    string urn = 1;                        // the Pulumi URN for this resource.
-    google.protobuf.Struct properties = 2; // the provider inputs to set during creation.
-    double timeout = 3;                    // the create request timeout represented in seconds.
-    bool preview = 4;                      // true if this is a preview and the provider should not actually create the resource.
-    string name = 5; // the Pulumi name for this resource.
-    string type = 6; // the Pulumi type for this resource.
+    // The URN of the resource being created.
+    string urn = 1;
+
+    // The resource's input properties, to be set during creation. These should have been validated by a call to
+    // [](pulumirpc.ResourceProvider.Check).
+    google.protobuf.Struct properties = 2;
+
+    // A timeout in seconds that the caller is prepared to wait for the operation to complete.
```

Comment:
Any preference on how the provider should indicate that the operation timed out?

Reply:
I need to look into what we actually do here. Will track in #17691